fairlearn/metrics/_metric_frame.py

```diff
@@ -382,19 +382,41 @@ def group_max(self) -> Union[Any, pd.Series, pd.DataFrame]:
         whether control features are present, and whether the metric functions
         were specified as a single callable or a dictionary.
 
+        Parameters
+        ----------
+        errors: {'ignore, 'raise', 'coerce'}, default 'raise'
+            if 'raise', then invalid parsing will raise an exception
+            if 'coerce', then invalid parsing will be set as NaN
+            if 'ignore', then invalid parsing will return the input
+
         Returns
         -------
         typing.Any or pandas.Series or pandas.DataFrame
             The maximum value over sensitive features. The exact type
             follows the table in :attr:`.MetricFrame.overall`.
         """
-        if not self.control_levels:
-            result = pd.Series(index=self._by_group.columns, dtype='object')
-            for m in result.index:
-                max_val = self._by_group[m].max()
-                result[m] = max_val
-        else:
-            result = self._by_group.groupby(level=self.control_levels).max()
+        if errors not in ("ignore", "raise", "coerce"):
+            raise ValueError("invalid error value specified")
+
+        try:
+            if not self.control_levels:
+                result = pd.Series(index=self._by_group.columns, dtype='object')
+                for m in result.index:
+                    max_val = self._by_group[m].max()
+                    result[m] = max_val
+            else:
+                result = self._by_group.groupby(level=self.control_levels).max()
+        # Catch ValueError that is thrown by pandas, if not all columns in metric frame are scalar
+        except ValueError:
+            warnings.warn(f"Metric frame contains non-scalar cells. "
+                          f"Please remove non-scalar columns from your metric frame and try again",
+                          SyntaxWarning)
+            if errors == "raise":
+                raise
+            elif errors == 'coerce':
+                return np.nan
```

Comment:
Don't these `else` clauses need to do different things based on control_levels? Also, how does this behave if some columns in the DataFrame can work with `max()` but others can't?

Reply:
Ah yes! That would be better behavior, that coerce would return all columns that do work with max, min, difference and ratio. Will update this